docs/testing/DEVELOPER-BEST-PRACTICES.md

```diff
@@ -0,0 +1,90 @@
+# Developer Best Practices
+## go:
+* mkdir -p $HOME/go/src/github.com/appsody
+* brew install go (mac only)
+## git:
+* brew install git (mac only)
+* ssh keys
+    * check for existing keys
+        * ls -la ~/.ssh
+            * look for key pairs, ex: id_rsa, id_rsa.pub
+        * if you don't have any keys you can generate new ones:
+            * cd ~/
+            * ssh-keygen -t rsa
+                * press enter to accept the defaults
+    * upload keys:
+        * copy key to clipboard
+            * pbcopy < ~/.ssh/id_rsa.pub
+        * log into github:
+            * https://github.com/settings/profile
+            * On the left-hand side menu, you will see a link “SSH and GPG keys”
+                * click "New SSH Key" button
+                * give it a name
+                * paste in the copied key from clipboard
+                * click "Add SSH Key" button
+## fork setup
+* create a fork for the appsody repo, ex:
+    * https://github.com/appsody/appsody
+        * click on the Fork button
+* clone your fork, ex:
+    * cd $HOME/go/src/github.com/appsody
+    * git clone git@github.com:tnixa/appsody.git
+* go into the repo
+    * cd appsody
+* verify you only have origin remote
+    * git remote -v
+* rename origin to your userid, ex:
+    * git remote rename origin tnixa
+ * add a remote for your fork, ex:
+    * git remote add upstream git@github.com:appsody/appsody.git
+* set master to point to upstream
+    * git branch master -u upstream/master
+## branch setup (every time you work on a new issue/PR)
+* checkout and update master
+    * git checkout master
+    * git pull master
+* create a branch, ex:
+    * git checkout -b testbranch
+* set tracking of branch
+    * make changes to the files you want
+* stage the changes
+    * git add .
+* commit the changes
+    * git commit -m "commit message"
+* push the changes and set tracking of the branch
+    * git push -u tnixa testbranch
```

Comment:
Maybe add a comment here that the `-u` and after is only needed the first time you push the branch. After that, `git push` is good enough.